app/src/main/java/cm/aptoide/pt/database/RoomNotificationPersistence.java

```diff
@@ -0,0 +1,80 @@
+package cm.aptoide.pt.database;
+
+import cm.aptoide.pt.database.room.NotificationDao;
+import cm.aptoide.pt.database.room.RoomNotification;
+import hu.akarnokd.rxjava.interop.RxJavaInterop;
+import io.reactivex.BackpressureStrategy;
+import io.realm.Sort;
+import java.util.List;
+import rx.Completable;
+import rx.Observable;
+import rx.Single;
+import rx.schedulers.Schedulers;
+
+public class RoomNotificationPersistence {
+
+  private NotificationDao notificationDao;
+
+  public RoomNotificationPersistence(NotificationDao notificationDao) {
+    this.notificationDao = notificationDao;
+  }
+
+  public Single<List<RoomNotification>> getDismissed(Integer[] notificationType, long startingTime,
+      long endTime) {
+    return RxJavaInterop.toV1Single(
+        notificationDao.getDismissed(notificationType, startingTime, endTime))
+        .subscribeOn(Schedulers.io());
+  }
+
+  public Single<List<RoomNotification>> getAllSorted(Sort sortOrder, Integer[] notificationType) {
+    return RxJavaInterop.toV1Single(notificationDao.getAllSortedDescByType(notificationType))
+        .subscribeOn(Schedulers.io());
+  }
+
+  public Single<RoomNotification> getLastShowed(Integer[] notificationType) {
+    return getAllSorted(Sort.DESCENDING, notificationType).map(notifications -> {
+      for (RoomNotification notification : notifications) {
+        if (!notification.isDismissed()) {
+          return notification;
+        }
+      }
+      return null;
```

Comment:
this should probably be a flatmap and instead of null Single.never()

Reply:
Changed it to a flatmap.
This is in Rx1 not rx2, so we dont have the single.never method